### PR TITLE
Add game populate

### PR DESCRIPTION
**Pull request by mmeitl**
Added game populate method and added tests for game populate.

### CODE REVIEW

File: spec/factories/piece_factory.rb [hunk 1]
```diff
@@ -6,5 +6,6 @@
     in_game true
     association :player
     association :game
+    after(:build) { |piece| piece.class.skip_callback(:create, :after, :populate_board!) }
```
Comment (mscoutermarsh) [inline spec/factories/piece_factory.rb]: You can remove this line. Now that you have the stub in the piece_spec

`allow_any_instance_of(Game).to receive(:populate_board!).and_return true`

File: app/spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# This file was generated by the `rspec --init` command. Conventionally, all
```
Comment (mscoutermarsh) [inline app/spec/spec_helper.rb:1]: is this file used anywhere? Or was it autogenerated?

If not used, can you delete it?

File: app/models/game.rb [hunk 3]
```diff
@@ -2,4 +2,17 @@ class Game < ActiveRecord::Base
...
+
+  def populate_board!
+    order_of_back_row = %w(Rook Knight Bishop King Queen Bishop Knight Rook)
+    order_of_back_row.each_with_index do |type, column|
+      Piece.create(type: type, game: self, color: 'black', row: 0, column: column)
+      Piece.create(type: type, game: self, color: 'white', row: 7, column: column)
+
+      Piece.create(type: 'Pawn', game: self, color: 'black', row: 1, column: column)
+      Piece.create(type: 'Pawn', game: self, color: 'white', row: 6, column: column)
+    end
```
Comment (mscoutermarsh) [inline app/models/game.rb:16]: love it.

File: app/spec/spec_helper.rb (hunk 2 above)
Comment (mmeitl) [inline app/spec/spec_helper.rb:1]: Hmm my copy of the file is not showing that so im not sure what to do

Comment (mmeitl) [inline app/spec/spec_helper.rb:1]: I can delete it if needed.

**Pull request merged**